dot/state/block.go

```diff
@@ -250,14 +250,62 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants even if hash is not stored in memory.
```

Comment:
nit

```suggestion
// GetAllDescendants gets all the descendants even if the block hash is not stored in memory.
```

Or maybe even

```suggestion
// GetAllDescendants gets all the descendants for a given block hash, by first checking in memory
// and, if not found, reading from disk.
```